decoder.tex

```diff
@@ -139,8 +140,8 @@ \section{Decoder pseudo code}
           #  final retired instruction
         inferred_address = TRUE
         return
-      if (te_inst.format == 3 and pc == address and
-        (branches == (is_branch(get_instr(pc)) ? 1 : 0)))
+      if (te_inst.format == 3 and pc == address and !unprocessed_branches(pc) and
+        (te_inst.privilege == privilege or is_return_from_trap(get_instr(last_pc))))

```

Comment:
Q: the process_te_inst initialized privilege to te_inst.privilege. This code then tests te_inst.privilege to be equal to privilege. Will not not be always true?

Reply:
No.  'privilege' is the global variable whereas te_inst.privilege is the privilege value from the current te_inst message being processed.  This value gets copied to the global variable 4 lines from the end of the code on page 82 of the PDF, whereas the compare is happening in the follow_execution_path call a few lines before this.

Reply:
Thanks. I misread the placement of the `privilege` initialization. Second question - should the `privilege` initialization be conditional based on `te_inst` being a format 3 packet?

Reply:
Last question about the initial state of the `privilege` global variable. It gets consumed before initialization on the first packet.

Reply:
No it doesn't.  follow_execution_path isn't called on the 1st packet because start_of_trace is TRUE.

Reply:
looks good to me.